lib/jekyll/commands/serve.rb

```diff
@@ -125,15 +125,14 @@ def mime_types
           WEBrick::HTTPUtils::load_mime_types(mime_types_file)
         end
 
-        def server_address(server, options)
-          baseurl = "#{options['baseurl']}/" if options['baseurl']
-          [
-            "http://",
-            server.config[:BindAddress],
-            ":",
-            server.config[:Port],
-            baseurl || ""
-          ].map(&:to_s).join("")
+        def server_address_info(server, options)
+          addr = server.config[:BindAddress].to_s
+          burl = "#{options['baseurl']}/" if options['baseurl']
+          ladr = addr == "0.0.0.0" ? "localhost" : a
+
+          rtn = "http://#{ladr}:#{server.config[:Port]}#{burl}"
+          rtn = "#{rtn} (listening on all interfaces)" if addr == "0.0.0.0"
+        rtn
```

Comment:
i'd prefer not to use `rtn`, but to name it something that specifies what it is, rather than what it'll be used for in this method.